extensions/resteasy-classic/resteasy/deployment/src/main/java/io/quarkus/resteasy/deployment/RestPathAnnotationProcessor.java

```diff
@@ -79,21 +81,29 @@ public void transform(TransformationContext ctx) {
                 MethodInfo methodInfo = target.asMethod();
                 ClassInfo classInfo = methodInfo.declaringClass();
 
-                AnnotationInstance annotation = methodInfo.annotation(REST_PATH);
-                if (annotation == null) {
-                    return;
+                AnnotationInstance classAnnotation = classInfo.classAnnotation(REST_PATH);
+                AnnotationInstance methodAnnotation = methodInfo.annotation(REST_PATH);
+                if (methodAnnotation == null) {
+                    // Check for @Path on class and not method
+                    if (!(classAnnotation != null && isRestEndpointMethod(methodInfo.annotations()))) {
```

Comment:
Is there any way to make this more readable? Lots of nots.. (that's a wish though.. )

Reply:
Hmm, I wonder if dropping the class annotation check completely will work, as that should still hit the case of method without `@Path` but with `@GET`. Will give it a try

Reply:
Ok, it works. Pushed the change